### PR TITLE
Feature/fetching indicator

### PR DESCRIPTION
**Pull request by manema**
### Fetching indicator and fetching error toast:

Description:
- WithApp HOC (Wraps all screens. Show an indicator when the app is performing an asynchronous action or a toast in case of a fetching error is received).
- Toast component.

Note: _fetchingCounter is a number and not a boolean because the rare case( but possible ) when an action is dispatched before the response of a previously dispatched action is handled. This would wrongly hide the fetching indicator when the application is still fetching._ 

query_counter

### CODE REVIEW

File: src/components/common/CustomToast/index.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import React, { Component } from 'react';
+import { bool, number, string, func } from 'prop-types';
+import Toast from 'react-native-easy-toast';
+
+const defaultDuration = 2000;
+
+class CustomToast extends Component {
+  constructor(props) {
+    super(props);
+    this.toasteRef = React.createRef();
```
Comment (felipegarcia92) [inline src/components/common/CustomToast/index.js]: typo: `toastRef`

File: src/components/common/WithApp/index.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import { node, number, string } from 'prop-types';
+import { View, ActivityIndicator } from 'react-native';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+import { clearFetchingError } from 'actions/appActions';
+import { getFetchingCounter, getError } from 'selectors/appSelector';
+import CustomToast from 'components/common/CustomToast';
+import styles from './styles';
+
+const WithApp = WrappedComponent => (props) => {
```
Comment (felipegarcia92) [inline src/components/common/WithApp/index.js]: Nice HOC! 

I see two naming issues here. First, HOCs normally start with lowercase, in this case, `withApp`.

Second, I think that `withApp` is waaay too generic. What about `withAsyncComponents` or something like that?

File: src/components/common/WithApp/index.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  return (
+    <View style={styles.container}>
+      <WrappedComponent {...props} />
+      { fetchingCounter !== 0 &&
+        <View style={styles.indicator}>
+          <ActivityIndicator size="large" />
+        </View>
+      }
+      { error &&
+      <CustomToast
```
Comment (felipegarcia92) [inline src/components/common/WithApp/index.js]: Indentation here, one more level in regard of the condition

```
{ condition &&
  <Component... 
}
```

File: src/constants/styleConstants.js [hunk 4]
```diff
@@ -9,6 +9,7 @@ export const SM = 360;
 export const BLACK = '#000000';
 export const WHITE = '#FFFFFF';
 export const RED = '#FF0000';
+export const TRANSPARENT = 'rgba(52, 52, 52, 0.8)';
```
Comment (peter7z) [inline src/constants/styleConstants.js]: `TRANSPARENT_GRAY` maybe?
Just transparents makes my think the color alpha is 0.

File: src/components/common/CustomToast/index.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  }
+
+  componentDidMount() {
+    const {
+      duration,
+      display,
+      message,
+      callbackOnFinish
+    } = this.props;
+    display && this.toastRef.current.show(message, duration || defaultDuration, callbackOnFinish);
```
Comment (agustito37) [inline src/components/common/CustomToast/index.js]: Default duration should be set in the default props.

```
CustomToast.defaultProps = {
  duration: defaultDuration,
};
```

**Pull request merged**